src/frontend/planner_test/tests/testdata/array.yaml

```diff
@@ -90,9 +90,6 @@
 - sql: |
     select array_append(123, 234);
   binder_error: 'Bind error: Cannot append integer to integer'
-- sql: |
```

Comment:
These should no longer throw an error, since we now allow the combination of two (multi-dimensional) arrays, as long as their nesting level is <= 1